cadc-vos-server-alt/src/main/java/org/opencadc/vospace/server/async/RecursiveDeleteNodeRunner.java

```diff
@@ -0,0 +1,358 @@
+
+/**
+ * Class to delete the target node and all child nodes recursively.
+ *
+ *
+ */
+public class RecursiveDeleteNodeRunner implements JobRunner {
+
+    private static Logger log = Logger.getLogger(RecursiveDeleteNodeRunner.class);
+
+    // containers are not locked while being emptied for the recursive delete and might need a few revisits in order
+    // for the operation to succeed.
+    private static final long PHASE_CHECK_INTERVAL = 1000; // 1 second
+    private static final long MAX_ERROR_BEFORE_ABORT = 100;
+
+    private Job job;
+    private JobUpdater jobUpdater;
+    private long lastPhaseCheck = System.currentTimeMillis();
+    private VOSpaceAuthorizer vospaceAuthorizer;
+    private NodePersistence nodePersistence;
+    private long deleteCount = 0;
+    private long errorCount = 0;
+    private JobLogInfo logInfo;
+    private String appName;
+
+    public URI resourceID;
+
+    public RecursiveDeleteNodeRunner() {
+    }
+
+    @Override
+    public void setAppName(String appName) {
+        String jndiNodePersistence = appName + "-" + NodePersistence.class.getName();
+        try {
+            Context ctx = new InitialContext();
+            this.nodePersistence = (NodePersistence) ctx.lookup(jndiNodePersistence);
+            this.vospaceAuthorizer = new VOSpaceAuthorizer(nodePersistence);
+        } catch (NamingException oops) {
+            throw new RuntimeException("BUG: NodePersistence implementation not found with JNDI key " + jndiNodePersistence, oops);
+        }
+    }
+
+    @Override
+    public void setJobUpdater(JobUpdater jobUpdater) {
+        this.jobUpdater = jobUpdater;
+    }
+
+    @Override
+    public void setJob(Job job) {
+        this.job = job;
+    }
+
+    @Override
+    public void setSyncOutput(SyncOutput syncOutput) {
+        // not used
+    }
+
+    @Override
+    public void run() {
+        log.debug("RUN RecursiveDeleteNodeRunner");
+        logInfo = new JobLogInfo(job);
+
+        String startMessage = logInfo.start();
+        log.info(startMessage);
+
+        long t1 = System.currentTimeMillis();
+        doit();
+        long t2 = System.currentTimeMillis();
+
+        logInfo.setElapsedTime(t2 - t1);
+
+        String endMessage = logInfo.end();
+        log.info(endMessage);
+    }
+
+    private void doit() {
+        try {
+            // set the phase to executing
+            ExecutionPhase ep = jobUpdater.setPhase(
+                job.getID(), ExecutionPhase.QUEUED, ExecutionPhase.EXECUTING, new Date());
+
+            if (ep == null) {
+                throw new IllegalStateException(
+                        "Could not change the job phase from " + ExecutionPhase.QUEUED
+                                + " to " + ExecutionPhase.EXECUTING);
+            }
+
+            VOSURI nodeURI = null;
+            for (Parameter param : job.getParameterList()) {
+                if (param.getName().equalsIgnoreCase("nodeURI")) {
```

Comment:
synctrans has a parameter to denote the target node and it's in the VOSpace-2.1 spec - what is the param name there? should use the same if it makes sense

also, container listing uses `uri` param as the "start at this one" so that's also a possibility

Reply:
Replaced with `target` is per async endpoint.